cdap-ui/app/cdap/components/Header/Welcome/index.tsx

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import * as React from 'react';
+import MyUserStoreApi from 'api/userstore';
+import If from 'components/If';
+import {Modal, ModalHeader, ModalBody} from 'reactstrap';
+import IconSVG from 'components/IconSVG';
+import Tour from 'components/Header/Tour';
+import {objectQuery} from 'services/helpers';
+import T from 'i18n-react';
+
+import './Welcome.scss';
+
+// Setting the store value to be an integer so that subsequent release, we can simply increment this number
+// to show the guided tour again.
+const USER_STORE_KEY = 'showWelcome';
+const USER_STORE_VALUE = 1;
+const SESSION_STORAGE_KEY = USER_STORE_KEY;
+const SESSION_STORAGE_VALUE = 'true';
+const PREFIX = 'features.NUX.Welcome';
+
+export default class Welcome extends React.PureComponent {
+  public state = {
+    showModal: false,
+    showAgain: false,
+  };
+
+  public componentDidMount() {
+    // Checking for session storage so that the welcome modal is not shown
+    // when user is navigating back and forth between react and angular pages.
+    const sessionValue = window.sessionStorage.getItem(SESSION_STORAGE_KEY);
+    if (sessionValue === SESSION_STORAGE_VALUE) { return; }
```

Comment:
nit: Shouldn't this be `false` if we don't want to show the welcome modal? The storage key is "showWelcome" but it value needs to be "false" right now for us to show the welcome modal (or "true" for us to not show the welcome modal?)

Reply:
good point. I'll change the value. As a side note, the value in sessionStorage has to be a string